src/structlog/processors.py

```diff
@@ -359,14 +373,69 @@ def format_exc_info(
 
     If there is no ``exc_info`` key, the *event_dict* is not touched.
     This behavior is analogue to the one of the stdlib's logging.
+
+    :param exception_formatter: A callable that is used to format the exception
+        from the ``exc_info`` field.
+
+    .. versionadded:: 22.1
     """
-    exc_info = event_dict.pop("exc_info", None)
-    if exc_info:
-        event_dict["exception"] = _format_exception(
-            _figure_out_exc_info(exc_info)
-        )
 
-    return event_dict
+    def __init__(
+        self,
+        exception_formatter: ProcessorExceptionFormatter = _format_exception,
+    ) -> None:
+        self.format_exception = exception_formatter
+
+    def __call__(
+        self, logger: WrappedLogger, name: str, event_dict: EventDict
+    ) -> EventDict:
+        exc_info = event_dict.pop("exc_info", None)
+        if exc_info:
+            event_dict["exception"] = self.format_exception(
+                _figure_out_exc_info(exc_info)
+            )
+
+        return event_dict
+
+
+format_exc_info = ExceptionFormatter()
+"""
+Replace an ``exc_info`` field with an ``exception`` string field using
+Python's built-in traceback formatting.
+
+If *event_dict* contains the key ``exc_info``, there are tree possible
+behaviors:
+
+- If the value is a tuple, render it into the key ``exception``.
+- If the value is an Exception render it into the key ``exception``.
+- If the value is true but no tuple, obtain exc_info ourselves and render
+    that.
```

Comment:
let's enumerate when we mention numbers:

```suggestion
1. If the value is a tuple, treat it like a tuple coming from `sys.exc_info`
   and render it into the key ``exception``.
2. If the value is an Exception, render it into the key ``exception``.
3. If the value is true but no tuple, obtain `sys.exc_info` ourselves and render
    that.
```